zabbix/stackstate_checks/zabbix/zabbix.py

```diff
@@ -288,25 +288,34 @@ def retrieve_problems(self, url, auth):
             acknowledged = item.get("acknowledged", None)
             # Object id is in case of object=0 a trigger.
             trigger_id = item.get("objectid", None)
-            # for Zabbix versions <4.0 we need to get the trigger.priority
-            severity = item.get("severity", self.get_trigger_priority(url, auth, trigger_id))
+
+            # for Zabbix versions <4.0 we need to get the trigger.priority and if priority doesn't exist then
+            # either trigger is disabled or doesn't exist
+            priority = self.get_trigger_priority(url, auth, trigger_id)
+            severity = item.get("severity", priority)
 
             zabbix_problem = ZabbixProblem(event_id, acknowledged, trigger_id, severity)
 
             self.log.debug("Parsed ZabbixProblem %s." % zabbix_problem)
             if not event_id or not trigger_id or not severity:
                 self.log.warn("Incomplete ZabbixProblem, got: %s" % zabbix_problem)
 
-            yield zabbix_problem
+            if priority is not None:
```

Comment:
What is the case when there is no priority available? Add a comment and/or unit test?

Reply:
if the trigger is enabled then for sure we will get the priority.

Reply:
put the comment for more clarification.